core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -1078,7 +1078,10 @@ private[spark] object JsonProtocol {
         val blockManagerAddress = blockManagerIdFromJson(json \ "Block Manager Address")
         val shuffleId = (json \ "Shuffle ID").extract[Int]
         val mapId = (json \ "Map ID").extract[Long]
-        val mapIndex = (json \ "Map Index").extract[Int]
+        val mapIndex = (json \ "Map Index") match {
+          case JNothing => 0
```

Comment:
So, this is only Spark History Server regression? Is there any side-effect on Spark 3.0.0?
> When Spark history server read event log produced

Reply:
Yes, only affects SHS

Reply:
Also created JIRA and updated title